File: javascript/169.majority-element.js
/*
 * @lc app=leetcode id=169 lang=javascript
 *
 * [169] Majority Element
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  let major = nums[0],
    count = 1;
  for (let i = 1; i < nums.length; i++) {
    if (count === 0) major = nums[i];
    if (major === nums[i]) count++;
    else count--;
  }
  return major;
};
// @lc code=end

File: rust/169.majority-element.rs
/*
 * @lc app=leetcode id=169 lang=rust
 *
 * [169] Majority Element
 */

// @lc code=start
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let (mut count, mut result) = (1, nums[0]);
        for i in 1..nums.len() {
            if count == 0 {
                result = nums[i];
            }
            match nums[i] == result {
                true => count += 1,
                false => count -= 1,
            }
        }
        return result;
    }
}
// @lc code=end
